src/megachatapi_impl.cpp

```diff
@@ -1768,6 +1768,27 @@ void MegaChatApiImpl::sendPendingRequests()
             });
             break;
         }
+        case MegaChatRequest::TYPE_IMPORT_MESSAGES:
+        {
+            if (mClient->initState() != karere::Client::kInitHasOfflineSession
+                            && mClient->initState() != karere::Client::kInitHasOnlineSession)
+            {
+                errorCode = MegaChatError::ERROR_ACCESS;
+                break;
+            }
+
+            int count = mClient->importMessages(request->getText());
+            if (count < 0)
+            {
+                API_LOG_WARNING("Error importing Messages: %d", count);
+                errorCode = MegaChatError::ERROR_UNKNOWN;
+                break;
+            }
+
+            API_LOG_DEBUG("%d Messages has been imported", count);
+            errorCode = MegaChatError::ERROR_OK;
```

Comment:
Caution! this request will never finish, since the `ERROR_OK == 0`, so no `fireOnRequestFinish()` would be called. I'll fix it ;)

Reply:
Fixed